trie/proof.go

```diff
@@ -505,7 +505,7 @@ func VerifyRangeProof(rootHash common.Hash, firstKey []byte, lastKey []byte, key
 		if val != nil || hasRightElement(root, firstKey) {
 			return false, errors.New("more entries available")
 		}
-		return hasRightElement(root, firstKey), nil
+		return false, nil
```

Comment:
This looks strange. With this change, this entire clause will always return `false` in the first return-parameter, I think

Reply:
Edit: nevermind, I think this is fine